index.d.ts

```diff
@@ -0,0 +1,168 @@
+  latency: number;
+  status: "disconnected" | "connecting" | "handshaking" | "ready" | "resuming";
+}
+
+interface ClusterStatsInfo {
+  guilds: number;
+  users: number;
+  totalRam: number;
+  voice: number;
+  exclusiveGuilds: number;
+  largeGuilds: number;
+  clusters: ClusterDataStats[];
+}
+
+interface ClusterDataStats extends Exclude<ClusterStatsInfo, "totalRam" | "clusters"> {
+  cluster: number;
+  shards: number;
+  ram: number;
+  uptime: number;
+  shardStats: ShardStats;
+}
+
+interface ClusterStats {
+  stats: ClusterStatsInfo;
+  clustersCounted: number;
+}
+
+interface ClusterData {
+  item: number;
+  value: {
+    id: number;
+    clusterCount: number;
+    name: string;
+    firstShardID: number;
+    lastShardID: number;
+    maxShards: number;
+    token: string;
+    file: string;
+    clientOptions: Eris.ClientOptions;
+  }
+}
+
+interface Webhook {
+  id: string;
+  token: string;
+}
+
+interface Webhooks {
+  shard?: Webhook;
+  cluster?: Webhook;
+}
+
+interface ClusterManagerOptions {
+  shards?: number;
+  firstShardID?: number;
+  lastShardID?: number;
+  clusters?: number;
+  clusterTimeout?: number;
+  stats?: boolean;
+  debug?: boolean;
+  statsInterval?: number;
+  name?: string;
+  guildsPerShard?: number;
+  webhooks?: Webhooks;
+  clientOptions: Eris.ClientOptions;
+}
+
+declare class IPC extends EventEmitter {
+  events: Map<string, { fn: Callback }>;
+  register(event: string, callback: Callback): void;
+  unregister(name: string): void;
+  broadcast(name: string, message?: any): void;
+  sendTo(cluster: number, name: string, message?: any): void;
+  fetchUser(id: string): Promise<Eris.User>;
+  fetchGuild(id: string): Promise<Eris.Guild>;
+  fetchChannel(id: string): Promise<Eris.AnyChannel>;
+  fetchMember(guildID: string, memberID: string): Promise<Eris.Member>;
+}
+
+declare class Cluster<T extends Base, C extends Eris.Client> {
+  shards: number;
+  maxShards: number;
+  firstShardID: number;
+  lastShardID: number;
+  mainFile: string | null;
+  clusterID: number;
+  clusterCount: number;
+  guilds: number;
+  users: number;
+  uptime: number;
+  exclusiveGuilds: number;
+  largeGuilds: number;
+  voiceChannels: number;
+  shardsStats: ShardStats[];
+  app: T | null;
+  bot: C;
+  test: boolean;
+  ipc: IPC;
+  logOverride(message: unknown): string;
+  spawn(): void;
+  connect(firstShardID: number, lastShardID: number, maxShards: number, token: string, type: unknown, clientOptions: Eris.ClientOptions): void;
+  loadCode(bot: C): void;
+  startsStats(bot: C): void;
+}
+
+class Queue extends EventEmitter {
+  queue: ClusterData[];
+  executeQueue(): void;
+  queueItem(item: ClusterData): void;
+}
+
+declare class ClusterManager<T extends Base, C extends Eris.Client> extends EventEmitter {
+  shardCount: number | "auto";
+  firstShardID: number;
+  lastShardID: number;
+  clusterCount: number;
+  clusterTimeout: number;
+  token: string | false;
+  clusters: Map<number, Cluster<T, C>>;
```

Comment:
clusters isn't actually a map of Cluster objects. Rather, it should be an object of {workerID: number, firstShardID: number, lastShardID: number}

Reply:
Is there not a way to access the cluster class then? The only reference I can find to it is in Cluster manager but that only gets used to spawn the cluster and doesn't get assigned to anything

Reply:
Indeed, there's no way to access the Cluster class from ClusterManager as they're created in different processes.